pages/index.tsx

```diff
@@ -1,64 +1,83 @@
-import { ChangeEvent, useEffect, useState } from "react";
-import { CrossCircleSVG, Spinner, Tag } from "@ensdomains/thorin";
+import { ChangeEvent, useState } from "react";
+import { CrossCircleSVG } from "@ensdomains/thorin";
 import { HomepageBg } from "@/components/01-atoms";
 import { isNameAvailable } from "@/lib/utils/blockchain-txs";
 import { ENSName, buildENSName } from "@namehash/ens-utils";
 import { DebounceInput } from "react-debounce-input";
 import { useRouter } from "next/router";
 import { normalize } from "viem/ens";
 import Link from "next/link";
+import { EnsDomainStatus } from "@/types/ensDomainStatus";
+import { domainWithEth, hasMoreThanOneDot } from "@/lib/utils/formats";
+import { EnsDomainStatusComponents } from "@/components/02-molecules";
 
 export default function Home() {
   const router = useRouter();
   const [domain, setDomain] = useState("");
-  const [isAvailable, setIsAvailable] = useState<boolean | null>(null);
-  const [domainError, setDomainError] = useState(false);
+  const [domainStatus, setDomainStatus] = useState<EnsDomainStatus>(
+    EnsDomainStatus.Available
+  );
 
-  useEffect(() => {
-    if (!domain) return;
+  const updateDomainStatus = (searchedDomain: string) => {
+    if (!searchedDomain) return EnsDomainStatus.Invalid;
 
-    setIsAvailable(null);
+    setDomainStatus(EnsDomainStatus.Searching);
 
     let ensName: null | ENSName = null;
+
+    // check if the domain is valid
     try {
-      ensName = buildENSName(domain);
+      normalize(searchedDomain);
+      ensName = buildENSName(searchedDomain);
     } catch (error) {
       console.error(error);
-      setIsAvailable(false);
+      setDomainStatus(EnsDomainStatus.Invalid);
       return;
     }
 
+    // check if domain is supported
+    if (hasMoreThanOneDot(domainWithEth(searchedDomain))) {
+      setDomainStatus(EnsDomainStatus.NotSupported);
+      return;
+    }
+
+    // check if domain is available
     isNameAvailable(ensName)
       .then((isAvailable: boolean) => {
-        setIsAvailable(isAvailable);
+        if (isAvailable) {
+          setDomainStatus(EnsDomainStatus.Available);
+        } else {
+          setDomainStatus(EnsDomainStatus.Registered);
+        }
       })
       .catch(() => {
-        setIsAvailable(false);
+        setDomainStatus(EnsDomainStatus.Registered);
       });
-  }, [domain]);
+  };
 
   const clearDomainSearch = () => {
     setDomain("");
   };
 
   const handleInputChange = (e: ChangeEvent<HTMLInputElement>) => {
-    setDomainError(false);
+    let searchedDomain = e.target.value.toLocaleLowerCase();
 
-    // handle errors
-    try {
-      normalize(e.target.value.trim().replace(" ", ""));
-    } catch (error) {
-      console.error(error);
-      setDomainError(true);
-    }
+    setDomain(searchedDomain);
+    updateDomainStatus(searchedDomain);
```

Comment:
I think it is better to make this function run everytime domain is updated, in case there is no problems doing so

Reply:
`handleInputChange` updates the domain and runs every time the input changes, so it is already being rendered each time the domain changes.